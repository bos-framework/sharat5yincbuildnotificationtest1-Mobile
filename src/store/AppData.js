import { AsyncStorage } from 'react-native';

class AppData {

  getItemForKey = async (key, callback) => {

    try {
      const value = await AsyncStorage.getItem(key);
      if (value !== null) {
        // We have data!!
        callback(true, value);
      }
    } catch (error) {
      // Error retrieving data
      callback(false, error.message);
    }

  };

  setItemForKey = async (key, value, callback) => {

    try {
      await AsyncStorage.setItem(key, value, () => {
        callback(true);
      });
    } catch (error) {
      callback(false);
    }
  }
  
}

export default new AppData();