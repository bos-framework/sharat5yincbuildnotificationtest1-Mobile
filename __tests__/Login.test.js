import React from 'react';
import LoginContainer from '../src/Screens/Container/loginContainer';
import Login from '../src/Screens/Presentational/login';
import store from '../src/store/store';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('Render Login component without crashing', () => {

  // Testing the presentational component render(Tracking expected/unexpected in UI).

  it('Render Login component without crashing', () => {
    const tree = renderer
      .create(<LoginContainer store={store} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });


  // Pass data to state of component.

  it('componentWillMount: set state values', () => {
    const wrapper = shallow(<LoginContainer store={store} />);
    wrapper.setState({ email: 'raghav@bosframework.com' })
    expect(wrapper.state('email')).toEqual('raghav@bosframework.com');
  });

  // Find the controls/elements/child component in the LoginContainer component.

  it('Find components/Elements', () => {

    const wrapper = shallow(<Login data={{
      email: '',
      password: '',
      errors: {
        emailEmpty: false,
        emailError: false,
        passwordEmpty: false,
        passwordError: false,
      },
      isLoading: false,
    }} />);
    expect(wrapper.find('Button').length).toEqual(2);
  });

  // Button action/press test. ToDo with all button controls.

  it('Button tap test', () => {

    let handleLoginMock = jest.fn()

    let LoginComponent = renderer.create(<Login onPressLogin={handleLoginMock}
      data={{
        email: '',
        password: '',
        errors: {
          emailEmpty: false,
          emailError: false,
          passwordEmpty: false,
          passwordError: false,
        },
        isLoading: false,
      }} />).getInstance()

    LoginComponent.props.onPressLogin();
    expect(handleLoginMock).toHaveBeenCalled()
  });

  it("should render a <View />", () => {
    expect(props.navigation.navigate).toHaveBeenCalledWith('LoginScreen');   // SUCCESS
  });

});




