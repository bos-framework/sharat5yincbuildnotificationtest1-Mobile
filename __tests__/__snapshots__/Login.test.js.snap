// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Login component without crashing Render Login component without crashing 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
      }
    }
  >
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={true}
      visible={false}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00000040",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "space-around",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#000",
              "borderRadius": 10,
              "display": "flex",
              "height": 75,
              "justifyContent": "space-around",
              "width": 75,
            }
          }
        >
          <ActivityIndicator
            animating={false}
            color="#999999"
            hidesWhenStopped={true}
            size="small"
          />
        </View>
      </View>
    </Modal>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.2,
          "justifyContent": "center",
          "marginTop": 10,
        }
      }
    >
      <Image
        resizeMode="center"
        source={
          Object {
            "testUri": "../../../src/Assets/Images/logo.png",
          }
        }
        styles={
          Object {
            "height": 160,
            "width": 160,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 0.8,
          "marginTop": 20,
          "maxWidth": 270,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderColor": "#bdbdbd",
              "borderRadius": 0,
              "borderWidth": 1,
              "marginTop": 10,
            },
            Object {
              "borderColor": "#bdbdbd",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          keyboardType="email-address"
          onChangeText={[Function]}
          placeholder="Email address"
          placeholderTextColor="#919090"
          rejectResponderTermination={true}
          returnKeyType="next"
          secureTextEntry={false}
          style={
            Object {
              "backgroundColor": "#dbdbdb",
              "borderRadius": 0,
              "color": "#545454",
              "fontSize": 16,
              "height": 44,
              "paddingBottom": 0,
              "paddingLeft": 10,
              "paddingTop": 2,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Array [
            Object {
              "borderColor": "#bdbdbd",
              "borderRadius": 0,
              "borderWidth": 1,
              "marginTop": 10,
            },
            Object {
              "borderColor": "#bdbdbd",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          keyboardType="default"
          onChangeText={[Function]}
          placeholder="Password"
          placeholderTextColor="#919090"
          rejectResponderTermination={true}
          returnKeyType="done"
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "#dbdbdb",
              "borderRadius": 0,
              "color": "#545454",
              "fontSize": 16,
              "height": 44,
              "paddingBottom": 0,
              "paddingLeft": 10,
              "paddingTop": 2,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#424142",
            "borderRadius": 2,
            "height": 44,
            "justifyContent": "center",
            "marginTop": 35,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 17,
            }
          }
        >
          Login
        </Text>
      </View>
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 2,
            "color": "#545454",
            "height": 44,
            "justifyContent": "center",
            "marginTop": 35,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 17,
              },
              Object {
                "color": "#000",
              },
            ]
          }
        >
          Forgot password?
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "color": "#fcfcfc",
          "height": 44,
          "justifyContent": "flex-start",
          "marginTop": 10,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#545454",
            "fontSize": 16,
          }
        }
      >
        Not Registered?
        <Text
          style={
            Array [
              Object {
                "color": "#545454",
                "fontSize": 16,
              },
              Object {
                "fontWeight": "600",
              },
            ]
          }
        >
           Create an account
        </Text>
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
